lib/collection/request.js

```diff
@@ -33,19 +33,25 @@ _.inherit((
         // this constructor is intended to inherit and as such the super constructor is required to be executed
         Request.super_.apply(this, arguments);
 
-        // in case definition object is missing, there is no point moving forward
-        if (!options) { return; }
+        // in case definition object is missing, create an empty one
+        options = options || {};
 
         // if the definition is a string, it implies that this is a get of URL
         (typeof options === 'string') && (options = {
             url: options
         });
 
+        this.update(options);
+    }), Property);
+
+_.assign(Request.prototype, /** @lends Request.prototype */ {
+
+    update: function (options) {
         _.mergeDefined(this, /** @lends Request.prototype */ {
             /**
              * @type {Url}
              */
-            url: _.createDefined(options, 'url', Url),
+            url: _.createDefined(options, 'url', Url, new Url()),
```

Comment:
no. this means, if a url is not defined when sent in update, even the existing url gets overwritten by empty one. :-/